### PR TITLE
Allow direct connection to a stack for ontology generation

### PR DESCRIPTION
**Pull request by mfedderly**
Adds the ability to generate an ontology directly from a given stack, given its stack URL and a clientId.
Users can still generate from a json file if they prefer.

### CODE REVIEW

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts [hunk 1]
```diff
@@ -180,6 +180,7 @@ export class ObjectSetListenerWebsocket<
     if (state) {
       const { subscriptionId, objectSet } = state;
       if (subscriptionId) {
+        state.subscriptionId = undefined;
```
Comment (mfedderly) [inline packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:183]: Unrelated cleanup

File: packages/cli/src/commands/typescript/generate/handleGenerate.mts [hunk 2]
```diff
@@ -14,18 +14,84 @@
...
+
+  const token = await invokeLoginFlow({
+    applicationId: clientId,
+    baseUrl: stack,
+    verbose: 0,
+  });
+  const { fetch } = createClientContext(
+    {
+      metadata: {
+        userAgent: "@osdk/cli/0.0.0 ()",
```
Comment (pashakondratyev) [inline packages/cli/src/commands/typescript/generate/handleGenerate.mts:62]: You can probably use `readCliVersionFromPackageJson` for now as well

**Review by pashakondratyev — APPROVED** [review]

File: packages/cli/src/commands/typescript/generate/TypescriptGenerateArgs.ts [hunk 3]
```diff
@@ -16,7 +16,9 @@
 
 export interface TypescriptGenerateArgs {
   outDir: string;
-  ontologyPath: string;
+  ontologyPath?: string;
+  stack?: string;
```
Comment (pashakondratyev) [inline packages/cli/src/commands/typescript/generate/TypescriptGenerateArgs.ts:20]: We should also clean up the other two args to use stack
